contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+/**
+ * @notice A contract to handle access control of subscription management dependent on signing a Terms of Service
+ */
+
+interface ITermsOfServiceAllowList {
+  /**
+   * @notice Check is the address is authorized for usage
+   * @param acceptor - The wallet address that has accepted the Terms of Service on the UI
+   * @param recipient - The recipient address that the acceptor is taking responsibility for
+   * @return Hash of the message data
+   */
+  function getMessageHash(address acceptor, address recipient) external pure returns (bytes32);
+
+  /**
+   * @notice Check is the address is authorized for usage
+   * @param messageHash - Message hash produced from "getMessageHash"
+   * @return Hash of the message data packed as "\x19Ethereum Signed Message\n" + len(msg) + msg
+   */
+  function getEthSignedMessageHash(bytes32 messageHash) external pure returns (bytes32);
+
+  /**
+   * @notice Check if the address is authorized for usage
+   * @param sender The transaction sender's address
+   * @return True or false
+   */
+  function isAllowedSender(address sender) external returns (bool);
+
+  /**
+   * @notice Check if the address is blocked for usage
+   * @param sender The transaction sender's address
+   * @return True or false
+   */
+  function isBlockedSender(address sender) external returns (bool);
+
+  /**
+   * @notice Allows access to the sender based on acceptance of the Terms of Service
+   * @param acceptor - The wallet address that has accepted the Terms of Service on the UI
+   * @param recipient - The recipient address that the acceptor is taking responsibility for
```

Comment:
I don't understand the recipient concept here. There's no mention of recipient in the Notion doc. Is the recipient always supposed to be a contract address, not an EoA? Or is the sub owner supposed to accept ToS on behalf of every downstream user...?

Reply:
Recipient can be either an EoA or a contract account.

Yeah, that's how I'd think about it - the sub owner is taking on responsibility for all of their dApp's users.